core/src/main/java/jenkins/telemetry/Telemetry.java

```diff
@@ -0,0 +1,196 @@
+    public static class TelemetryReporter extends AsyncPeriodicWork {
+
+        public TelemetryReporter() {
+            super("telemetry collection");
+        }
+
+        @Override
+        public long getRecurrencePeriod() {
+            return TimeUnit.HOURS.toMillis(24);
+        }
+
+        @Override
+        protected void execute(TaskListener listener) throws IOException, InterruptedException {
+            if (UsageStatistics.DISABLED || !Jenkins.getInstance().isUsageStatisticsCollected()) {
+                LOGGER.info("Collection of anonymous usage statistics is disabled, skipping telemetry collection and submission");
+                return;
+            }
+            Telemetry.all().forEach(telemetry -> {
+                if (telemetry.getStart().isAfter(LocalDate.now())) {
+                    LOGGER.config("Skipping telemetry for '" + telemetry.getId() + "' as it is configured to start later");
+                    return;
+                }
+                if (telemetry.getEnd().isBefore(LocalDate.now())) {
+                    LOGGER.config("Skipping telemetry for '" + telemetry.getId() + "' as it is configured to end in the past");
+                    return;
+                }
+
+                JSONObject data = new JSONObject();
+                try {
+                    data = telemetry.createContent();
+                } catch (Exception e) {
+                    LOGGER.log(Level.WARNING, "Failed to build telemetry content for: '" + telemetry.getId() + "'", e);
+                }
+
+                JSONObject wrappedData = new JSONObject();
+                wrappedData.put("type", telemetry.getId());
+                wrappedData.put("payload", data);
+                wrappedData.put("correlator", ExtensionList.lookupSingleton(Correlator.class).getCorrelationId());
+
+                try {
+                    URL url = new URL(ENDPOINT);
+                    URLConnection conn = ProxyConfiguration.open(url);
+                    if (!(conn instanceof HttpURLConnection)) {
+                        LOGGER.config("URL did not result in an HttpURLConnection: " + ENDPOINT);
+                        return;
+                    }
+                    HttpURLConnection http = (HttpURLConnection) conn;
+                    http.setRequestProperty("Content-Type", "application/json; charset=utf-8");
+                    http.setDoOutput(true);
+
+                    String body = wrappedData.toString();
+                    if (LOGGER.isLoggable(Level.FINEST)) {
+                        LOGGER.finest("Submitting JSON: " + body);
+                    }
+
+                    try (OutputStreamWriter writer = new OutputStreamWriter(http.getOutputStream(), StandardCharsets.UTF_8)) {
+                        writer.append(body);
+                    }
+
+                    LOGGER.config("Telemetry submission received response '" + http.getResponseCode() + " " + http.getResponseMessage() + "' for: " + telemetry.getId());
```

Comment:
To get better performance, use `LOGGER.log(Level.CONFIG, ...)` . Resolving strings for CONFIG-level logging is not fun

Reply:
It's one string concatenation per day and collector/trial, I think this is not a substantial load..